dags/oup/parser.py

```diff
@@ -34,3 +39,19 @@ def _get_dois(self, article: ET.Element):
         self.logger.msg("Parsing dois for article", dois=doi)
         self.dois = doi
         return [doi]
+
+    def _get_journal_doctype(self, article: ET.Element):
+        self.logger.msg("Parsing journal doctype for article", dois=self.dois)
+        journal_doctype_raw = article.find("front/..").get("article-type")
+        if not journal_doctype_raw:
+            return
+        try:
+            journal_doctype = self.article_type_mapping[journal_doctype_raw]
+        except KeyError:
+            journal_doctype = "other"
+        if "other" in journal_doctype:
```

Comment:
does it happen anytime except of the `KeyError`? If no, maybe you can remove `if` statement

Reply:
Yes, just when the type is not found in article type mapping. If the the type is None, this parsing step has to crash

Reply:
Can it be a different exception in this case?

Reply:
no, I would just move logger warning to the exception handling and remove `if` clause